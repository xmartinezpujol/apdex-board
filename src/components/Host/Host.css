.host-container {
    background-color: #FFF;
    padding: 30px;
    margin: 15px 0px;
}

.host-container strong {
    color: #000;
    font-size: 16px;
    font-family: "HelveticaNeue-Bold", Helvetica, Arial, sans-serif;
    line-height: 19px;
}

.host-container.list {
    max-width: 100%;
}

.host-container.grid {
    width: 375px;
}

@media only screen and (max-width: 991px) {
    .host-container.grid {
        width: 100%;
    }
}
