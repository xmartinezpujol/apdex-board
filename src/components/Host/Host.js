import Component from './../Component';
import AppItem from './../AppItem/AppItem';

import appConfig from './../../config/Constants';

import './Host.css';

class Host extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.name,
      apps: props.apps,
      appItems: [],
    };
    this.init();
  }

  init() {
    this.state.appItems = this.state.apps.map(item => (new AppItem(item)));
  }

  render(layoutMode = 'grid') {
    return `
      <div class="host-container ${layoutMode}">
        <strong>${this.state.name}</strong>
          ${this.state.appItems.slice(0, appConfig.MAX_APPS_BY_HOST).map(app => app.render()).join('')}
      </div>
   `;
  }
}

export default Host;
