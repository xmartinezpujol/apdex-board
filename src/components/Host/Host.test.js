import Host from './Host';
import mockHost from '../../data/MockHost';
import appConfig from './../../config/Constants';

const host = new Host(mockHost);

document.body.innerHTML = host.render();

describe('Host', () => {
  it(`has a name`, () => {
    expect(host.state.name).toBeDefined();
  });

  it(`should have in memory ${appConfig.MAX_APPS_APDEX_DESC} Apps (max)`, () => {
    expect(host.state.appItems.length)
      .toBeLessThanOrEqual(appConfig.MAX_APPS_APDEX_DESC);
  });

  it(`should render only ${appConfig.MAX_APPS_BY_HOST} Apps (max)`, () => {
    expect(document.querySelectorAll('div.app-item').length)
      .toBeLessThanOrEqual(appConfig.MAX_APPS_BY_HOST);
  });

  it('renders correctly', () => {
    expect(host.render('grid')).toMatchSnapshot();
  });

  it('should adapt list style when layout is changed', () => {
    document.body.innerHTML = host.render('list');
    expect(document.getElementsByClassName('host-container')[0].classList.contains('list'))
      .toBe(true);
  });
});
