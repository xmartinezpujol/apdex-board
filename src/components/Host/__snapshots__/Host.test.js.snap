// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host renders correctly 1`] = `
"
      <div class=\\"host-container grid\\">
        <strong>7e6272f7-098e.dakota.biz</strong>
          
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[1].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">100</div>
        <p>Awesome 1</p>
      </div>
    
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[2].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">99</div>
        <p>Awesome 2</p>
      </div>
    
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[3].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">98</div>
        <p>Awesome 3</p>
      </div>
    
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[4].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">96</div>
        <p>Awesome 4</p>
      </div>
    
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[5].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">95</div>
        <p>Awesome 5</p>
      </div>
    
      </div>
   "
`;
